src/image.rs

```diff
@@ -299,6 +305,22 @@ impl<'docker> Images<'docker> {
             .try_flatten_stream(),
         )
     }
+
+    /// Deletes unused images
+    ///
+    /// [Api Reference](https://docs.docker.com/engine/api/v1.42/#tag/Image/operation/ImagePrune)
+    pub async fn prune(
+        &self,
+        opts: &PruneOptions,
+    ) -> Result<String> {
+        let mut path = vec!["/images/prune".to_string()];
+
+        if let Ok(Some(query)) = opts.serialize() {
+            path.push(query)
+        }
+
+        self.docker.post(&path.join("?"), None).await
```

Comment:
I don't like manually constructing the URL here by doing a manual path join instead of using a proper URL library or at least having functions for this, but this matches what the rest of the shiplift code is doing, so this is fine.